### PR TITLE
DAOS-2503 test: Enable mdtest in soak

### PR DESCRIPTION
**Pull request by mjean308**
Quick-Functional: true
Skip-unit-tests: true
Skip-func-test-vm: true
Skip-func-hw-test-small: true
Skip-func-hw-test-medium: true
Test-tag-hw-large: soak_smoke

Signed-off-by: Maureen Jean <maureen.jean@intel.com>

### CODE REVIEW

**Review by shimizukko — DISMISSED** [review]

**Review by dinghwah — DISMISSED** [review]
LGTM

File: src/tests/ftest/soak/soak_harassers.yaml [hunk 1]
```diff
@@ -166,6 +161,36 @@ fio_harasser:
...
+        - 4
+        - 16
+    test_dir: "/"
+    api:
+        - 'DFS'
+        - 'POSIX'
+    iteration: 1
+    num_of_files_dirs: 10000
+    pre_iter: 10
+    flags: '-L -F -N 1 -P -v'
```
Comment (saurabhtandan) [inline src/tests/ftest/soak/soak_harassers.yaml:181]: Options '-L' and '-F' will ensure the operations only happen on files which might be good for a performance test but I believe for a soak test we should include directory operations as well.
I am not quite sure how '-N 1' impacts results, but if it is something to get optimized numbers, may be we can remove that too.

File: src/tests/ftest/soak/soak_smoke.yaml [hunk 2]
```diff
@@ -155,6 +151,32 @@ daos_racer:
...
+    taskspernode:
+        - 1
+    test_dir: "/"
+    api:
+        - 'DFS'
+        - 'POSIX'
+    iteration: 1
+    num_of_files_dirs: 100
+    pre_iter: 10
+    flags: '-L -F -N 1 -P -v'
```
Comment (saurabhtandan) [inline src/tests/ftest/soak/soak_smoke.yaml:168]: Same as above.

File: src/tests/ftest/soak/soak_stress_2h.yaml [hunk 3]
```diff
@@ -118,4 +112,30 @@ ior_stress:
...
+    taskspernode:
+        - 1
+    test_dir: "/"
+    api:
+        - 'DFS'
+        - 'POSIX'
+    iteration: 1
+    num_of_files_dirs: 1000
+    pre_iter: 10
+    flags: '-L -F -N 1 -P -v'
```
Comment (saurabhtandan) [inline src/tests/ftest/soak/soak_stress_2h.yaml:130]: Same as above.

File: src/tests/ftest/soak/soak_stress_48h.yaml [hunk 4]
```diff
@@ -152,11 +143,40 @@ fio_stress:
...
+    taskspernode:
+        - 16
+    test_dir: "/"
+    api:
+        - 'DFS'
+        - 'POSIX'
+    iteration: 1
+    num_of_files_dirs: 3000
+    branching_factor: 1
+    flags: '-L -F -N 1 -P -v'
```
Comment (saurabhtandan) [inline src/tests/ftest/soak/soak_stress_48h.yaml:167]: Same as above

File: src/tests/ftest/soak/soak_utils.py [hunk 5]
```diff
@@ -719,6 +749,94 @@ def create_ior_cmdline(self, job_spec, pool, ppn, nodesperjob):
...
+
+    Returns:
+        cmd: cmdline string
+
+    """
+    commands = []
+    mdtest_params = "/run/" + job_spec + "/"
+    mpi_module = self.params.get(
+        "mpi_module", "/run/*", default="mpi/mpich-x86_64")
+    # IOR job specs with a list of parameters; update each value
```
Comment (saurabhtandan) [inline src/tests/ftest/soak/soak_utils.py:771]: Minor: typo IOR=MDTEST

File: src/tests/ftest/soak/soak_utils.py [hunk 6]
```diff
@@ -719,6 +749,94 @@ def create_ior_cmdline(self, job_spec, pool, ppn, nodesperjob):
...
+    api_list = self.params.get("api", mdtest_params + "*")
+    write_bytes_list = self.params.get("write_bytes", mdtest_params + "*")
+    read_bytes_list = self.params.get("read_bytes", mdtest_params + "*")
+    depth_list = self.params.get("depth", mdtest_params + "*")
+    flag = self.params.get("flags", mdtest_params + "*")
+    oclass_list = self.params.get("dfs_oclass", mdtest_params + "*")
+    num_of_files_dirs = self.params.get(
+        "num_of_files_dirs", mdtest_params + "*")
+    # update IOR cmdline for each additional IOR obj
```
Comment (saurabhtandan) [inline src/tests/ftest/soak/soak_utils.py:780]: Minor: typo IOR=MDTEST

**Review by shimizukko — APPROVED** [review]

**Review by saurabhtandan — APPROVED** [review]
LGTM

**Comment by mjean308** [comment]
Failing test is due to DAOS-7350

**Pull request merged**